### PR TITLE
Bug/properly configure ballots

### PR DESCRIPTION
**Pull request by DevArtemNuss**

### CODE REVIEW

File: src/store/trails/actions.js [hunk 1]
```diff
@@ -193,6 +195,51 @@ export const addBallot = async function ({ commit, state, rootState }, ballot) {
...
+      } else {
+        togglebal = {
+          account: 'telos.decide',
+          name: 'togglebal',
+          data: {
+            ballot_name: ballotName,
+            setting_name: ballot.config
+          }
+        }
+        actions.splice(2, 0, togglebal)
```
Comment (karynemayer) [inline src/store/trails/actions.js:240]: Would it be possible to apply a single `actions.splice(2, 0, togglebal)` at line 242? I think that way, we could focus on defining `togglebal` from line 192 up to 242, and then assign it to `actions`. Do you think it makes sense?

**Review by karynemayer — CHANGES_REQUESTED** [review]
@DevArtemNuss you did an amazing job understanding all the new concepts from EOSIO blockchain and the requirements for fixing this bug. ❤️

I left a question for you and a few suggestions for the code. Any comment I left you feel it doesn't make sense, you're more than welcome to question it and make your own suggestions.

File: src/pages/trails/ballots/components/ballot-form.vue [hunk 2]
```diff
@@ -51,8 +53,19 @@ export default {
...
         .map(treasury => ({
           label: treasury.title ? `${treasury.title} (${treasury.supply})` : treasury.supply,
-          value: treasury.supply
+          value: treasury.supply,
+          symbol: treasury.supply.replace(/[^a-zA-Z]/gi, '')
         }))
+    },
+    isStakeble () {
+      if (this.form.treasurySymbol && this.form.treasurySymbol.symbol) {
+        return this.treasuries.find(t => (t.access === 'public' || t.manager === this.account) && t.symbol === this.form.treasurySymbol.symbol).settings.find(i => i.key === 'stakeable').value
```
[the remark this answers is not in the document]
Comment (DevArtemNuss) [reply src/pages/trails/ballots/components/ballot-form.vue:62]: I check settings for current treasury and take config for this

[the remark this answers is not in the document]
Comment (karynemayer) [reply src/pages/trails/ballots/components/ballot-form.vue]: I see!
My only suggestion for readability would be to break this return statement a bit.

```
let selectedTreasurySettings = this.treasuries.find(t => (t.access === 'public' || t.manager === this.account) && t.symbol === this.form.treasurySymbol.symbol)?.settings
return selectedTreasurySettings.find(i => i.key === 'stakeable')?.value
```

Adding the `?` will protect us from throwing an error and breaking the application if, for some obscure reason, we don't get any result from the `find` function.

**Review by karynemayer — APPROVED** [review]
🙌

File: src/store/trails/actions.js [hunk 3]
```diff
@@ -193,6 +195,49 @@ export const addBallot = async function ({ commit, state, rootState }, ballot) {
...
+          }
+        }
+      } else {
+        togglebal = {
+          account: 'telos.decide',
+          name: 'togglebal',
+          data: {
+            ballot_name: ballotName,
+            setting_name: ballot.config
+          }
```
Comment (DonaldPeat) [inline src/store/trails/actions.js:235]: It looks like all of the `togglebal ` objects are the same except for `setting_name`, you _could_ add a method to return that object passing that value.

**Review by DonaldPeat — CHANGES_REQUESTED** [review]

File: src/store/trails/actions.js [hunk 4]
```diff
@@ -193,6 +202,28 @@ export const addBallot = async function ({ commit, state, rootState }, ballot) {
...
+      togglebal.data.setting_name = 'voteliquid'
+    } else if (ballot.settings && ballot.config) {
+      if (ballot.config === 'both') {
+        for (let i of ['voteliquid', 'votestake']) {
+          togglebal = {
+            account: 'telos.decide',
+            name: 'togglebal',
+            data: {
+              ballot_name: ballotName,
+              setting_name: i
```
Comment (DonaldPeat) [inline src/store/trails/actions.js:217]: you can use the same object here just setting the `setting_name`

**Review by DonaldPeat — CHANGES_REQUESTED** [review]

**Review by DonaldPeat — APPROVED** [review]

**Pull request merged**